### DISCUSSION TITLE
Add reload keybind

### DISCUSSION THREAD

**Discussion by FlareFlo** [FakeFishGames/Barotrauma#9522] [Ideas and suggestions]
This should take the proper ammo out of inventory/toolbelt and move it into the currently held weapon

**Comment by TopDogRolly** [comment]
Can be done currently, but its a bit inconvenient

**Reply by FlareFlo** [reply]
How so?

**Reply by TopDogRolly** [reply]
When you hold a weapon (its equipped) and you press a corresponding button (1, 2, 3 etc.) that holds a magazine or bullets, they will be loaded into the weapon.

**Reply by FlareFlo** [reply]
That is very good to know though, I will bind an extra mouse button to a specific Hotbar slot to fill that gap until a dedicated button is available!

**Reply by Krzeszny** [reply]
> This should take the proper ammo out of inventory/toolbelt and move it into the currently held weapon

Manual reloading makes up for the lack of a reloading animation. Since you have a hotbar, it's already easy if you're prepared.

Adding a keybind only adds problems without solving any (what happens if you have many types of ammo?)

**Reply by FlareFlo** [reply]
> Adding a keybind only adds problems without solving any (what happens if you have many types of ammo?)

Many games I've played let one press and hold the reload key to choose the target ammo to load for short presses.

**Reply by Krzeszny** [reply]
> > Adding a keybind only adds problems without solving any (what happens if you have many types of ammo?)
> 
> Many games I've played let one press and hold the reload key to choose the target ammo to load for short presses.

Does any of those games allow you to place bullets and magazines on the hotbar and when you press a hotkey, you reload the ammo under that key?

**Reply by FlareFlo** [reply]
> Does any of those games allow you to place bullets and magazines on the hotbar and when you press a hotkey, you reload the ammo under that key?
Not that I remember.

**Reply by Krzeszny** [reply]
On second thought, it wouldn't be complex to code **without new UI**. It's just a matter of making a keybind automatically load any ammo that fits.

But right now reloading requires skill and preparation, which adds immersion. 
I've always thought that reloading with one button (making it as simple as pulling the trigger) is even more immersion-breaking than the lack of safe/semi selectors.
